### PR TITLE
Support Coco style annotations

### PR DESCRIPTION
**Pull request by zytx121**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by liuyanyi — APPROVED** [review]
LGTM

File: configs/_base_/datasets/dota_coco.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# dataset settings
+dataset_type = 'mmdet.CocoDataset'
+data_root = 'data/split_ms_dota/'
+file_client_args = dict(backend='disk')
+
+train_pipeline = [
+    dict(type='mmdet.LoadImageFromFile', file_client_args=file_client_args),
+    dict(type='mmdet.LoadAnnotations', with_bbox=True, with_mask=True),
```
Comment (RangiLyu) [inline configs/_base_/datasets/dota_coco.py:8]: Can we set poly2mask=False to use polygon mask as default? It will be faster than bitmap mask.

Comment (RangiLyu) [inline configs/_base_/datasets/dota_coco.py:8]: We can set poly2mask=False to load polygon mask. There is no need to convert the annotation to bitmap mask and then convert it to rbox.

File: configs/_base_/datasets/dota_coco.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+]
+
+metainfo = dict(
+    CLASSES=('plane', 'baseball-diamond', 'bridge', 'ground-track-field',
+             'small-vehicle', 'large-vehicle', 'ship', 'tennis-court',
+             'basketball-court', 'storage-tank', 'soccer-ball-field',
+             'roundabout', 'harbor', 'swimming-pool', 'helicopter'))
+
+train_dataloader = dict(
+    batch_size=1,
```
Comment (RangiLyu) [inline configs/_base_/datasets/dota_coco.py:45]: The default batch size is 1?

File: configs/_base_/datasets/dota_coco.py (hunk 1 above)
Comment (zytx121) [reply configs/_base_/datasets/dota_coco.py:8]: Very reasonable!

**Pull request merged**